feat: TFLint plugin manager (#17954)
renovatebot/renovate: lib/modules/manager/tflint-plugin/readme.md

> This will maintain the [TFLint configuration file](https://github.com/terraform-linters/tflint/blob/master/docs/user-guide/config.md), and renovates the plugins it contains.

Reviewer: ```suggestion
Renovate maintains your [TFLint configuration file](https://github.com/terraform-linters/tflint/blob/master/docs/user-guide/config.md), and updates the plugins in the file.
```

How about this?

I don't like the non-standard verb `renovates`, we should use `updates` instead as that is a word every software engineer understands. 😉

Author: Agree, I picked up that wording from another readme but I also found it weird :)

Reviewer: > I picked up that wording from another readme but I also found it weird :)

@multani Can you give me a link to that other readme?

This will maintain the [TFLint configuration
file](https://github.com/terraform-linters/tflint/blob/master/docs/user-guide/config.md),

> and renovates the plugins it contains.

Reviewer 2: Keep this all on a single markdown line

> Supports plugins from github.com only.

Reviewer: Do we need to explain if the TFLint manager supports private/public repositories on `github.com`?

Reviewer 2: Technically it can support either if the token has permissions (so same as any other manager). I'm not sure if TFlint has a need for private or not though

Author: It doesn't support plugins in private repositories at the moment (and I don't think it will ever do, AFAIU)

Author: Sounds good :+1: 

I'll let you check if you also want to add something like "this is a limitation of TFLint itself".

Reviewer: Yeah, let's explain _why_ we don't support private repositories. 😉 

```suggestion
Supports plugins hosted in _public_ repositories on github.com.
This is because TFLint only supports _public_ repositories.
```
